### PR TITLE
ref: Continue if individual manifest fails

### LINKED ISSUE #21: Continue processing other manifests, even if one fails
Even if one or more manifests fail to be parsed, we should still continue to process others. This is improvement after adding better error reporting in #20 to address #13 

This will require some ref. around handling errors, I imagine the collector should return multiple "non-fatal" errors in this case, and these should be logged to user.

### PR DESCRIPTION
**Pull request by david-doit-intl**
Closes #21 

Signed-off-by: david-doit-intl <david@doit-intl.com>

### CODE REVIEW

**Comment by david-doit-intl** [comment]
@stepanstipl  I thought about change the function signature for Get() =>

from:
```go
type Collector interface {
	Get() ([]map[string]interface{}, error)
	Name() string
}
```
to
```go 
type Collector interface {
	Get() []map[string]interface{}
	Name() string
}
```

however there maybe some errors we wish to return which are not related to manifest parsing.

File: pkg/collector/file.go [hunk 1]
```diff
@@ -48,7 +48,8 @@ func (c *FileCollector) Get() ([]map[string]interface{}, error) {
 			input, err = ioutil.ReadFile(f)
 		}
 		if err != nil {
-			return nil, fmt.Errorf("failed to read file %s: %v", f, err)
+			log.Info().Msgf("failed to read file %s: %v", f, err)
+			continue
```
Comment (stepanstipl) [inline pkg/collector/file.go:52]: If a file, that was explicitly passed via flag by the user, is not found, we should fail. I wouldn't change this - this is equivalent to failing to connect to the cluster in other collectors (or similar).

File: pkg/collector/file.go [hunk 2]
```diff
@@ -73,7 +74,8 @@ func (c *FileCollector) Get() ([]map[string]interface{}, error) {
 				var manifest map[string]interface{}
 				err := yaml.Unmarshal([]byte(manifests[k]), &manifest)
 				if err != nil {
-					return nil, fmt.Errorf("failed to parse file %s: %v", f, err)
+					log.Info().Msgf("failed to parse file %s: %v", f, err)
```
Comment (stepanstipl) [inline pkg/collector/file.go:77]: Maybe `Warning` is in place?

File: pkg/collector/helm2.go [hunk 3]
```diff
@@ -78,8 +78,8 @@ func (c *HelmV2Collector) Get() ([]map[string]interface{}, error) {
 
 			err := yaml.Unmarshal([]byte(m), &manifest)
 			if err != nil {
-				err := fmt.Errorf("failed to parse release %s/%s: %v", r.Namespace, r.Name, err)
-				return nil, err
+				log.Info().Msgf("failed to parse release %s/%s: %v", r.Namespace, r.Name, err)
```
Comment (stepanstipl) [inline pkg/collector/helm2.go:81]: Warning?

File: pkg/collector/helm3.go [hunk 4]
```diff
@@ -77,8 +77,8 @@ func (c *HelmV3Collector) Get() ([]map[string]interface{}, error) {
 
 			err := yaml.Unmarshal([]byte(m), &manifest)
 			if err != nil {
-				err := fmt.Errorf("failed to parse release %s/%s: %v", r.Namespace, r.Name, err)
-				return nil, err
+				log.Info().Msgf("failed to parse release %s/%s: %v", r.Namespace, r.Name, err)
```
Comment (stepanstipl) [inline pkg/collector/helm3.go:80]: Warning?

**Review by stepanstipl — COMMENTED** [review]
Thanks 👍 .

Some minor tweaks - in general, what about using `Warn` level messages, to be consistent with existing code + it seems to be in place.

However, the PR is missing the main reason for the original #21 - when the `last-applied-configuration` can't be parsed as valid json, please add (in cluster collector - https://github.com/doitintl/kube-no-trouble/blob/055df4e213a394ad107021724714c5130759b284/pkg/collector/cluster.go#L125-L126).

**Comment by david-doit-intl** [comment]
@stepanstipl updated

File: pkg/collector/file_test.go [hunk 5]
```diff
@@ -93,12 +89,10 @@ func TestFileCollectorGetNonExistent(t *testing.T) {
 		t.Errorf("Expected to succeed for %s, failed: %s", input, err)
 	}
 
-	_, err = c.Get()
+	result, _ := c.Get()
```
Comment (stepanstipl) [inline pkg/collector/file_test.go:92]: I think this test should be reverted as well.

**Review by stepanstipl — COMMENTED** [review]
Looks good 👍 , I think you forgot to revert one test .. so just sort that one.

**Comment by stepanstipl** [comment]
LGTM 👍

**Review by stepanstipl — APPROVED** [review]
LGTM 👍

**Pull request merged**